tests/testem_connection_tests.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const patchEmitterForWildcard = require('../public/testem/testem_connection');
+const expect = require('chai').expect;
+const ServerIO = require('socket.io');
+const ClientIO = require('socket.io-client');
+
+describe.only('Testem Connection', function() {
+  var serverIo, client;
+
+  function replaceGlobals(newGlobals, originalGlobals) {
+    for (let key in newGlobals) {
+      originalGlobals[key] = global[key];
+      global[key] = newGlobals[key];
+    }
+  }
+  
+  before(function() {
+    var globals = {};
+    serverIo = ServerIO();
+    serverIo.listen(8000);
+  
+    replaceGlobals({
+      io: ClientIO
+    }, globals);
+
+    serverIo.on('connection', function (socket) {
+      socket.emit('foo', { bar: 'baz' })
+    });
+  });
+
+  after(function() {
+    serverIo.close();
+  });
+
+  afterEach(function() {
+    client.close();
+  });
+  
+  it('patches emitter for wildcard', function(done) {
+    client = ClientIO('http://localhost:8000');
+    patchEmitterForWildcard(client);
+
+    var eventNameArr = [];
+
+    function check(eventName) {
+      eventNameArr.push(eventName);
+      if(eventNameArr.length === 2) {
+        expect(eventNameArr).to.deep.equal(['*', 'foo']);
+        done();
```

Comment:
If the above expect fails, does the test suite error report correctly? At one point, mocha would simply let the test timeout here. If you could double check that it does not (maybe share the test output) that would be great.

Reply:
Check if the test failed correctly by changing a value in the array. 
here is the output of test failing purposely.
```
1) Testem Connection patches emitter for wildcard:

      Uncaught AssertionError: expected [ '*', 'foo' ] to deeply equal [ '*', 'baz' ]
      + expected - actual

       [
         "*"
      -  "foo"
      +  "baz"
       ]
```